common/telemetry/grpc.go

```diff
@@ -48,12 +53,49 @@ func NewServerTraceInterceptor(
 	tp trace.TracerProvider,
 	tmp propagation.TextMapPropagator,
 ) ServerTraceInterceptor {
-	return ServerTraceInterceptor(
-		otelgrpc.UnaryServerInterceptor(
-			otelgrpc.WithPropagators(tmp),
-			otelgrpc.WithTracerProvider(tp),
-		),
+	//nolint:staticcheck
+	otelInterceptor := otelgrpc.UnaryServerInterceptor(
```

Comment:
`UnaryServerInterceptor` is deprecated. But using the new function will take some significant effort I'd rather not put in right now. (I abandoned my initial attempt after 5 minutes)